server/interfaces/interfaces.go

```diff
@@ -449,6 +451,18 @@ type MetricsCollector interface {
 	ReadCount(ctx context.Context, counterName string) (int64, error)
 }
 
+// A KeyValStore allows for storing ephemeral values globally.
+//
+// No guarantees are made about durability of KEyValStores -- they may be
+// evicted from the backing store that maintains them (usually memcache or
+// redis), so they should *not* be used in critical path code.
+type KeyValStore interface {
+	SetByKey(ctx context.Context, key string, val []byte) error
+	GetByKey(ctx context.Context, key string) ([]byte, error)
```

Comment:
uber nit: these could just be `Get/Set` (since most functionality afforded by a key-value store requires a single key as input) -- also harmonious with `redis.Get` / `redis.Set`

Reply:
They can't be, since redis_cache is one of the things that needs to implement this interface, and Get and Set are already defined for the digest-style Cache, which it implements. First thing I tried :/

Reply:
I think `redis_cache` is maybe not the best place to implement this KV interface -- I get that it's convenient because `redis_cache` already owns a Redis client and has some Redis-related logic already in there, but our notion of "cache" today usually refers to the CAS and the Action Cache (AC), so I think it may be confusing to have this generalized KV interface implemented by something that is really intended to just implement the CAS and AC. For example, `disk_cache` is another cache implementation, but I'd guess we won't want a KV implementation that is disk-based.

I think it's fine to create a separate one-off implementation, maybe `enterprise/server/util/redis_key_value_store.go`